systemd/systemdtest/systemdtest.go

```diff
@@ -59,3 +63,57 @@ Type=simple
 	}
 	return output
 }
+
+type FakeSystemd struct {
```

Comment:
unless we plan to very soon switch to this all the current systemd mocking tests (except anyway we can't switch systemd package own tets), this feels a bit overkill and also not quite enough because of the issue mentioned in the other comment. My preference for now would be do something smaller and mostly deal with snapstate/backend for now.

We should have a internal helper in systemd that can be used by all 3 systemd.New*

something like:

`func newSystemd(kind string, rootDir string, mode InstanceMode, rep reporter)`  where kind is "emulation" or "full"

and have a systemd.MockNewSystemd helper there as well. With this, one can hijack at the same time any of the New* methods. Then the tests in backend can have a local implementation of systemd.Systemd. We can have minimal implementation perhaps dealing only with the methods we care about doing:

```
type FakeSystem struct {
   systemd.Systemd
   ...
}

methods...
```

Reply:
I see, it does make sense. Just a couple of considerations:

1. Is it OK if the `kind` is an enumerated type, instead of `string`?
2. We already have a `systemdtest` package, maybe the new `MockNewSystemd` helper could live there (though I'm not sure, if this is possible)